lib/semantics/resolve-names.cpp

```diff
@@ -1483,6 +1484,9 @@ Attrs AttrsVisitor::GetAttrs() {
   CHECK(attrs_);
   return *attrs_;
 }
+bool AttrsVisitor::HasPointerOrAllocatable() {
```

Comment:
This surprises me.  To be clear, there are three places in resolve-names.cpp where we test to see if the POINTER or ALLOCATABLE attribute is present.  You're saying that it would be better to repeat the code in these three instances than to have a separate function?

Reply:
What is the benefit of wrapping that simple predicate up in a function?

Reply:
It reduces the size of the code.

If you'd rather eliminate this function, I'm happy to do that.  It seemed wrong to me to copy/paste existing code rather than sharing it.  I'll remove the function.

Reply:
Defining the function increases the number of lines of code by 3, yes?  That's not a big deal, but it's surely not a reduction in code size.  Is there some other benefit?

Reply:
Not that I know of.